src/System.Numerics.Tensors/src/System/Numerics/Tensors/IDenseTensor.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Numerics.Tensors
+{
+    public interface IDenseTensor<T> : ITensor<T>
+    {
+        Memory<T> Buffer { get; }
```

Comment:
I'm not convinced returning `Memory<T>` is the right thing here.

I think, given this is a property, returning just `Span<T>` is likely sufficient. The fact that it is internally a `Memory<T>` probably doesn't need to be exposed.

Reply:
That's a good point, Memory is more advanced than Span.  If folks need a non-stack bound object they can hold onto the tensor.

Reply:
Would we want to turn it into a method then?

That way folks aren't doing things like:

```C#
for (int i = 0; i < tensor.Buffer.Length; i++)
{
    sum +=  tensor.Buffer[i];
}
```

I find it hard to justify having a property for things that should be cached in a local variable. I now that's how `Memory<T>` was designed (it has a Span property), but I still find that confusing.

Reply:
Yes, I think exposing this a method would be more beneficial. Maybe `AsSpan()` (which matches what we do in some other places)?